express-error-handling/server.ts

```diff
@@ -0,0 +1,143 @@
+/* eslint-disable @typescript-eslint/no-unused-vars -- REMOVE ME */
+import { readFile, writeFile } from 'node:fs/promises';
+import express from 'express';
+import { ClientError } from './client-error.js';
+import { errorMiddleware } from './error-middleware.js';
+
+type Note = {
+  id: number;
+  content: string;
+};
+type Data = {
+  nextId: number;
+  notes: Record<string, Note>;
+};
+
+const app = express();
+app.use(express.json());
+
+async function readData(): Promise<Data> {
+  const data = await readFile('./data.json');
+  return JSON.parse(data.toString());
+}
+
+async function writeData(data: Data) {
+  await writeFile('./data.json', JSON.stringify(data, null, 2));
+}
+
+app.get('/api/notes', async (req, res, next) => {
+  try {
+    const data = await readData();
+    const notes: Note[] = [];
+    for (const id in data.notes) {
+      notes.push(data.notes[id]);
+    }
+    res.json(notes);
+  } catch (err) {
+    next(err);
+    console.error(err);
```

Comment:
Delete lines 38 & 39, also, since `next` is doing them for you.
Do this in all your endpoints.